src/v/storage/storage_resources.cc

```diff
@@ -187,7 +187,9 @@ storage_resources::get_falloc_step(std::optional<uint64_t> segment_size_hint) {
         // we don't roll segments until they overshoot the size.
         step = std::min(step, segment_size_hint.value() + _append_chunk_size);
     }
-    return step;
+
+    // Allocate at least one chunk's worth of space.
+    return std::max(step, _append_chunk_size);
```

Comment:
should we look to see if there are any clusters out there that have a configured chunk size would be too large? i see the max in the configuration.cc is 32mb, but that seems rediculous. but even smaller values like 1mb, that are a lot larger than what i ewould expect in practice (like 32kb) could make this dynamic fallocation optimization fail for cases where we have huge numbers of partitions?